cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go

```diff
@@ -428,6 +438,23 @@ func isInstanceStatusNotRunningYet(instanceStatus string) bool {
 	return instanceStatus == "" || instanceStatus == "PROVISIONING" || instanceStatus == "STAGING"
 }
 
+func isInvalidReservationError(errorCode, errorMessage string) bool {
+	reservationErrors := []string{
+		"Incompatible AggregateReservation VMFamily",
+		"Could not find the given reservation with the following name",
+		"must use ReservationAffinity of",
+		"The reservation must exist in the same project as the instance",
+		"it requires reservation to be in READY state",
```

Comment:
Can you maybe create a separate error code for this? When backing off, it might be useful to differentiate between incorrect vs unready reservation...

Reply:
Separated the unready reservation message into another error code